### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by J-pilon**
Main update is the addition of a Pagination class

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -1,16 +1,22 @@
 class ArticlesController < ApplicationController
     def index
-        pageNumber = permittedParam[:page]&.to_i || 1
-        @articles = Article.all.order(created_at: :desc).limit(10)
-        @lastPage = (Article.count / 10) + 1
-
-        if (pageNumber < 1 || pageNumber > @lastPage)
+        if permittedParam[:page] == "0"
```
Comment (aarondufall) [inline app/controllers/articles_controller.rb:3]: I don't think an error is even needed here, I would just redirect to page 1.
```
redirect_to articals_path(page: 1)
```

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
-        else
-            pageNumber = (pageNumber - 1)
-            @articles.offset(10 * pageNumber)
+            @articles = Article.all.order(created_at: :desc)
+            @last_page = 1
+
+            return render :index
         end
+        
+        pag = Pagination.new(
```
Comment (aarondufall) [inline app/controllers/articles_controller.rb:11]: I don't think a class is the best way to build a reusable version of this feature. 
What other options are available to you what are the trade-offs? I would discuss this over discord or on the next call.

File: app/javascript/components/Paginator.jsx [hunk 3]
```diff
@@ -5,39 +5,58 @@ const Paginator = (props) => {
 
     const { link, lastPage } = props;
 
-    const pageNumber = parseInt(new URLSearchParams(window.location.search).get("page"));
+    const itemsLimitParam = parseInt(new URLSearchParams(window.location.search).get("items_limit")) || 0;
+    const pageNumber = parseInt(new URLSearchParams(window.location.search).get("page")) || 1;
```
Comment (aarondufall) [inline app/javascript/components/Paginator.jsx:9]: I would consider a utility function that will fetch all the search params and put them in an object.

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (J-pilon) [reply app/controllers/articles_controller.rb:3]: Cool. I had the same feeling

File: app/javascript/components/Paginator.jsx (hunk 3 above)
Comment (J-pilon) [reply app/javascript/components/Paginator.jsx:9]: cool cool. that makes sense.

**Review by rhurtis — COMMENTED** [review]
It looks good, I can't see anything wrong.

**Pull request merged**